### PR TITLE
[#2274] Introduce the source API

### LINKED ISSUE #2274: Source API
**Is your feature request related to a problem? Please describe.**
As part of the custom sources milestone (see for motivation), we need an app that exposes a common interface for all custom source connectors. 

**Describe the solution you'd like**
A Kafka streaming component that exposes the following functionality:

- [x] Manage source token permissions
- [x] Create sources
- [x] Ingest messages and metadata
- [x] Create channels
- [x] React to outbound messages

**Additional context**

All interactions with this app need to be authenticated so that you can run custom source apps from outside the cluster, too. For this purpose "creating a source" means: 
- Provision a JWT that encodes the source identifier and does not expire so that the source app can use it as an API key for interactions
- Create the necessary webhook (to be done by the source app)

### PR DESCRIPTION
**Pull request by chrismatix**
resolves #2274 

TODO:
- [x] Polish docs
- [x] Source api token authentication scheme
- [x] Add helm chart
- [x] Create source
- [x] Create channel
- [x] Ingestion webhook
- [x] Call action endpoint

### CODE REVIEW

File: frontend/chat-plugin/lib/src/AiryChatPlugin.tsx [hunk 1]
```diff
@@ -21,24 +24,10 @@ export const AiryChatPlugin = (props: AiryChatPluginProps) => {
 
   window.addEventListener('resize', handleResize);
 
-  const widgetHeight = (): number => {
```
Comment (chrismatix) [inline frontend/chat-plugin/lib/src/AiryChatPlugin.tsx:24]: cc @AitorAlgorta

File: docs/docs/getting-started/installation/configuration.md [hunk 2]
```diff
@@ -127,6 +127,8 @@ components:
   integration:
     webhook:
       name: webhook
+    source-api:
+      enabled: true
```
Comment (ljupcovangelski) [inline docs/docs/getting-started/installation/configuration.md:131]: This should be under `components.api`, but we don't have such a section yet, correct?

**Review by ljupcovangelski — APPROVED** [review]
The deployment charts and the docs look good :+1:

**Review by AitorAlgorta — APPROVED** [review]
Really good work, also with the some nice refactoring <3

**Review by ljupcovangelski — APPROVED** [review]
Very nice :+1:

**Pull request merged**